services/execution/service.go

```diff
@@ -477,6 +478,9 @@ func (s *Service) writeContent(ctx context.Context, mediaType, ref string, r io.
 }
 
 func (s *Service) getTask(ctx context.Context, id string) (plugin.Task, error) {
+	if _, err := namespaces.NamespaceRequired(ctx); err != nil {
+		return nil, grpc.Errorf(codes.InvalidArgument, "task %v not found: %v", id, err.Error())
```

Comment:
Would have to make a special case of the first call to Get then, since it's called in a loop for all runtimes (granted, we only have one atm).

If namespaces are always needed, maybe we should enforce it via an interceptor?